sklearn/ensemble/ensemble_selection.py

```diff
@@ -0,0 +1,261 @@
+        return ens_score, ens_proba
+
+    def _build_ensemble(self, X, y, ranked_estimators_name):
+        ensemble = Counter(ranked_estimators_name[:self.n_best])
+        ens_score, ens_proba = self._get_score_of_ensemble(X, y, ensemble)
+        n_clfs = sum(ensemble.values())
+
+        cand_ensembles = []
+        while(n_clfs < self.max_bag_estimators):
+            new_ens_scores = []
+            for name, new_clf in self.estimators:
+                new_clf_proba = self.clfs_proba_[name]
+
+                new_ens_proba = ((ens_proba * n_clfs + new_clf_proba)
+                                 / (n_clfs + 1))
+                new_ens_score = self.score_metric_(y, new_ens_proba)
+                new_ens_scores.append({'score':new_ens_score,
+                                       'new_estimator_name':name})
+
+            best_ens_score = max(new_ens_scores, key=lambda x: x['score'])
+
+            ens_score = best_ens_score['score']
+            new_estimator_name = best_ens_score['new_estimator_name']
+            ensemble.update({new_estimator_name: 1})
+
+            # store current ensemble to select best later
+            ens_copy = Counter(ensemble)
+            cand = {'ens': ens_copy, 'score': ens_score}
+            cand_ensembles.append(cand)
+
+            n_clfs = sum(ensemble.values())
+
+        if (n_clfs == self.max_bag_estimators):
+            best_cand_ensemble = max(cand_ensembles, key = lambda x: x['score'])
+            ensemble = best_cand_ensemble['ens']
+
+        return ensemble
+
+    def _get_final_ensemble(self, X, y):
+
+        n_clfs = int(len(self.estimators) * (1 - self.prune_fraction))
+        bag_size = int(self.bag_fraction * n_clfs)
+
+        ranked_clfs_score = sorted(self.clfs_score_.items(),
+                                   key=operator.itemgetter(1), reverse=True)
+        ranked_clfs_name = map(lambda (name, clf_score): name,
+                                     ranked_clfs_score)
+
+        if (self.verbose):
+            print "Ranked Model:"
+            for (i, model_score) in enumerate(ranked_clfs_score):
+                print "%d. " % i, model_score
+
+        bag_ensembles = []
+        rs = check_random_state(self.random_state)
+        for i in xrange(self.n_bags):
+            # get bag_size elements at random
+            cand_indices = sorted(rs.permutation(n_clfs)[:bag_size])
+            cand_ranked_clfs_name = [ranked_clfs_name[ci]
+                                     for ci in cand_indices]
+
+            bag_ensemble = self._build_ensemble(X, y, cand_ranked_clfs_name)
+            bag_ensembles.append(bag_ensemble)
+
+        # combine each ensemble built by bag ensemble selection
+        self.ensemble_ = Counter()
+        for bag_ensemble in bag_ensembles:
+            self.ensemble_ += bag_ensemble
+
+        # average the result of each bag ensemble selection
+        for clf in self.ensemble_:
+            self.ensemble_[clf] /= float(self.n_bags)
```

Comment:
I'm not sure it's appropriate to still be using a `Counter` here. Just copy it into a dict.